utilities/plugins/openai_plugin.py

```diff
@@ -0,0 +1,225 @@
+import json
+import time
+import requests
+import urllib3
+from typing import Any, Optional, Dict
+from urllib3.exceptions import InsecureRequestWarning
+from simple_logger.logger import get_logger
+
+urllib3.disable_warnings(category=InsecureRequestWarning)
+requests.packages
+LOGGER = get_logger(name=__name__)
+
+
+class OpenAIClient:
+    """
+    A client for interacting with the OpenAI API.
+
+    Attributes:
+        host (str): The base URL for the API.
+        streaming (bool): Flag to indicate if streaming requests should be used.
+        model_name (str, optional): The name of the model to use.
+        request_func (Callable): The function to use for making requests.
+    """
+
+    def __init__(self, host: Any, streaming: bool = False, model_name: Any = None) -> None:
+        """
+        Initializes the OpenAIClient.
+
+        Args:
+            host (str): The base URL for the API.
+            streaming (bool, optional): If True, use streaming requests. Defaults to False.
+            model_name (str, optional): The name of the model to use. Defaults to None.
+        """
+        self.host = host
+        self.streaming = streaming
+        self.model_name = model_name
+        self.request_func = self.streaming_request_http if streaming else self.request_http
+
+    def request_http(self, endpoint: str, query: Dict[str, str], extra_param: Optional[Dict[str, Any]] = None) -> Any:
+        """
+        Sends a HTTP POST request to the specified endpoint and processes the response.
+
+        Args:
+            endpoint (str): The API endpoint to send the request to.
+            query (dict): The query parameters to include in the request.
+            extra_param (dict, optional): Additional parameters to include in the request.
+
+        Returns:
+            Any: The parsed response from the API.
+
+        Raises:
+            pytest.Fail: If the request fails due to an exception.
+        """
+        headers = {"Content-Type": "application/json"}
+        data = self._construct_request_data(endpoint, query, extra_param)
+        try:
+            url = f"{self.host}{endpoint}"
+            time.sleep(5)  # sometime while making multiple request it fails
```

Comment:
Can we avoid using sleeps? Maybe consider using `tenacity` if you need to retry when a request fails